### PR TITLE
Try to fix bug(?) while render centered sprite

### PR DESCRIPTION
**Pull request by yardsoul**
I tried to use function provided by @davidgs which is awesome but I think it might not work as intended and will work only while it is on top of the screen since it always render entire sprite at 0,0 and using sprite canvas to center it. ( I am new to this so please correct me if I am wrong)

### CODE REVIEW

**Comment by davidgs** [comment]
So as I understand it, when you are writing to a Sprite, you write to `(0,0)` which is relative to the Sprite, and then you push the Sprite to where you want it `spr.pushSprite(x, y);` 

If you look at the analogous function: 
```cpp
void KGFX::drawText(TFT_eSprite &spr, const char *txt, const tftfont_t &f, int color, int x, int y) {
  tft.TTFdestination(&spr);
  spr.fillSprite(TFT_BLACK);

  tft.setTTFFont(f);
  tft.setTextColor(color, TFT_BLACK);
  tft.setCursor(0,0);
  tft.print(txt);

  spr.pushSprite(x, y);
}
```
Which is why the `drawTextCenter()`  function is:
```cpp
void KGFX::drawTextCenter(TFT_eSprite &spr, const char *txt, const tftfont_t &f, int color, int y)
{
  tft.TTFdestination(&spr);
  spr.fillSprite(TFT_BLACK); // Do not fill sprite with black, it will overwrite the sprite

  tft.setTTFFont(f);
  tft.setTextColor(color, TFT_BLACK);
  int x = tft.width() / 2 - tft.TTFtextWidth(txt) / 2;
  tft.setCursor(x, y);
  tft.print(txt);

  spr.pushSprite(0, 0);
}
```
(in looking at this, the Sprite should probably be pushed to `spr.pushSprite(0,y)` so I'll make that change.)

You will see that it is done the same way. Now, if you want to write text to a specific location on the screen, you wouldn't use a Sprite, but instead would use the analog to 
```cpp
void KGFX::drawText(const char *txt, const tftfont_t &f, int color, int x, int y) {
  tft.setTTFFont(f);
  tft.setTextColor(color, TFT_BLACK);
  tft.setCursor(x,y);
  tft.print(txt);
}
```
And the `Center()` version of that: 
```cpp
void KGFX::drawTextCenter(const char *txt, const tftfont_t &f, int color, int y)
{
  tft.setTTFFont(f);
  tft.setTextColor(color, TFT_BLACK);
  ttf.drawTextCentre(txt, 0, y, tft.width(), tft.height());
  int x = tft.width() / 2 - tft.TTFtextWidth(txt) / 2;
  tft.setCursor(x, y);
  tft.getFont();
  tft.print(txt);
}
```

**Comment by audreylim** [comment]
Setting the x position at 0 requires a 240 width sprite so that the text is centered, thereby using up more memory than needed. We can do something like this:

```
void KGFX::drawTextCenter(TFT_eSprite &spr, const char *txt, const tftfont_t &f, int color, int y)
{
  tft.TTFdestination(&spr);
  spr.fillSprite(TFT_BLACK);
  
  tft.setTTFFont(f);
  tft.setTextColor(color, TFT_BLACK);

  int w = tft.TTFtextWidth(txt); 
  int h = tft.TTFtextWidth(txt);
  int x1 = (spr.width() - w)/2; // center text in sprite
  int x2 = (tft.width() - spr.width)/2; // center sprite on screen
  int y1 = (spr.height() - h)/2; --> (we can either enforce the y axis at the top of the sprite, or center it. This line centers it.)

  tft.setCursor(x1, y1);
  tft.print(txt);

  spr.pushSprite(x2, y);
 }
```

This centers the text in the sprite, then centers the sprite on the screen.

**Comment by yardsoul** [comment]
> Setting the x position at 0 requires a 240 width sprite so that the text is centered, thereby using up more memory than needed. We can do something like this:
> 
> ```
> void KGFX::drawTextCenter(TFT_eSprite &spr, const char *txt, const tftfont_t &f, int color, int y)
> {
>   tft.TTFdestination(&spr);
>   spr.fillSprite(TFT_BLACK);
>   
>   tft.setTTFFont(f);
>   tft.setTextColor(color, TFT_BLACK);
> 
>   int w = tft.TTFtextWidth(txt); 
>   int h = tft.TTFtextWidth(txt);
>   int x1 = (spr.width() - w)/2; // center text in sprite
>   int x2 = (tft.width() - spr.width)/2; // center sprite on screen
>   int y1 = (spr.height() - h)/2; --> (we can either enforce the y axis at the top of the sprite, or center it. This line centers it.)
> 
>   tft.setCursor(x1, y1);
>   tft.print(txt);
> 
>   spr.pushSprite(x2, y);
>  }
> ```
> 
> This centers the text in the sprite, then centers the sprite on the screen.

I agree with your method but I removed y1 from center y in the sprite to make it align with other center method. If we want it to be center, we should agree to modify both methods.

File: kgfx.cpp [hunk 1]
```diff
@@ -69,15 +69,20 @@ void KGFX::drawText(TFT_eSprite &spr, const char *txt, const tftfont_t &f, int c
...
-  spr.fillSprite(TFT_BLACK); // Do not fill sprite with black, it will overwrite the sprite
+  spr.fillSprite(TFT_BLACK);
 
   tft.setTTFFont(f);
   tft.setTextColor(color, TFT_BLACK);
-  int x = tft.width() / 2 - tft.TTFtextWidth(txt) / 2;
-  tft.setCursor(x, y);
+
+  int w = tft.TTFtextWidth(txt); 
+  int h = tft.TTFtextWidth(txt);
```
Comment (audreylim) [inline kgfx.cpp]: In that case we don't need `h` anymore.

**Comment by davidgs** [comment]
LGTM

**Review by audreylim — APPROVED** [review]

**Pull request merged**